### PR TITLE
improve store configuration view

### PR DESCRIPTION
**Pull request by ccarruitero**
In order to get a better integration with recently solidus versions, I extracted the store form fields added for this gem that are missing in backend into a new partial and inject into template with deface.

And only load the other store views for more old solidus versions.

before
without_fix
after
with_fix

### CODE REVIEW

**Comment by ccarruitero** [comment]
@jhawthorn @tvdeyen can you take a look on this?

**Comment by ccarruitero** [comment]
@kennyadsl can you take a look into this PR?

This PR fix missing fields in store view when use this gem. The spec failing in master is not related to the changes in the PR.

File: lib/spree_multi_domain/engine.rb [hunk 1]
```diff
@@ -55,5 +55,10 @@ def current_order_with_multi_domain(options = {})
     initializer 'spree.promo.register.promotions.rules' do |app|
       app.config.spree.promotions.rules << Spree::Promotion::Rules::Store
     end
+
+    # only load store views if solidus < 2.2
+    if SolidusSupport.solidus_gem_version < Gem::Version.new('2.2.x')
```
Comment (kennyadsl) [inline lib/spree_multi_domain/engine.rb]: I think we can remove this support since 2.1 is out of EOL. WDYT?

**Comment by aitbw** [comment]
Hey @ccarruitero, can we get a rebase in here? :raised_hands:

**Pull request closed without merging**